drivers/video/video.c

```diff
@@ -3331,8 +3444,18 @@ static int video_complete_capture(uint8_t  err_code, uint32_t datasize)
         }
       else
         {
-          g_video_data_ops->set_buf((FAR uint8_t *)container->buf.m.userptr,
-                                    container->buf.length);
+          if (container->buf.memory == V4L2_MEMORY_MMAP)
```

Comment:
why do you need check V4L2_MEMORY_MMAP? userptr already point to the right place.